node/pkg/guardiansigner/amazonkms.go

```diff
@@ -0,0 +1,167 @@
+package guardiansigner
+
+import (
+	"bytes"
+	"context"
+	"crypto/ecdsa"
+	"encoding/asn1"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/kms"
+	kms_types "github.com/aws/aws-sdk-go-v2/service/kms/types"
+	"github.com/aws/aws-sdk-go/aws"
+	ethcrypto "github.com/ethereum/go-ethereum/crypto"
+)
+
+var (
+	secp256k1N     = ethcrypto.S256().Params().N
+	secp256k1HalfN = new(big.Int).Div(secp256k1N, big.NewInt(2))
+)
+
+type asn1EcSig struct {
+	R asn1.RawValue
+	S asn1.RawValue
+}
+
+type asn1EcPublicKey struct {
+	EcPublicKeyInfo asn1EcPublicKeyInfo
+	PublicKey       asn1.BitString
+}
+
+type asn1EcPublicKeyInfo struct {
+	Algorithm  asn1.ObjectIdentifier
+	Parameters asn1.ObjectIdentifier
+}
+
+type AmazonKms struct {
+	KeyId  string
+	Region string
+	svc    *kms.Client
+}
+
+func NewAmazonKmsSigner(unsafeDevMode bool, keyPath string) (*AmazonKms, error) {
+	amazonKmsSigner := AmazonKms{
+		KeyId: keyPath,
+	}
+
+	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithRegion("eu-north-1"))
+	if err != nil {
+		return nil, errors.New("failed to load default config")
+	}
+
+	amazonKmsSigner.svc = kms.NewFromConfig(cfg)
+
+	return &amazonKmsSigner, nil
+}
+
+func (a *AmazonKms) Sign(hash []byte) (signature []byte, err error) {
```

Comment:
I'm sorry if I failed to flag this before, but i think it would be ideal if both `Sign` and `Verify` were to have a `context.Context` parameter.

Reply:
Do you think I can just pass in the contexts that are available in the calling scope, and create a new one in scopes that don't have an existing context available?

Reply:
It's better than having no Context so I'm good with it.